include/nbl/builtin/hlsl/blit/temp.hlsl

```diff
@@ -12,8 +12,10 @@ namespace hlsl
 
 namespace spirv
 {
+// DXC fails with uint32_t2 result, so we have to use a struct.
+struct umul_result_t { uint32_t lsb; uint32_t msb; };
 [[vk::ext_instruction(/* OpUMulExtended */ 151)]]
-uint32_t2 umulExtended(uint32_t v0, uint32_t v1);
+umul_result_t umulExtended(uint32_t v0, uint32_t v1);
```

Comment:
doesn't HLSL have a builtin function for this?

Reply:
I also thought that it has because DXBC has such instruction from long time ago, but it seems they never exposed it to HLSL:
https://github.com/microsoft/DirectXShaderCompiler/issues/2821

Reply:
hmm, btw we now require `64bit` integers, so is that an issue any more?

Reply:
For this particular use case it's not a problem since the operation can safely be replaced by normal multiplication of two `uint64_t`s. I can use 64-bit ints to avoid having to do this, I left this because I wanted to be as close to the GLSL code as possible. It's only a problem if it's in performance critical code where the difference between `OpIMul` on 64 bit integers and `OpUMulExtended` might matter or if one actually needs the msb of a 64 bit multiplication. In this case this only one multiplication is tiny part of the computational cost of the whole shader...

Reply:
yes, it was a hack to not require 64bit integer support, but after upgrading our minimum to Vulkan 1.3 and supported devices (those that still get drivers) + requiring BDA, we've basically had to require 64bit int support too